rclcpp/include/rclcpp/node.hpp

```diff
@@ -1559,6 +1559,10 @@ class Node : public std::enable_shared_from_this<Node>
    * which has been created using an existing instance of this class, but which
    * has an additional sub-namespace (short for subordinate namespace)
    * associated with it.
+   * A suboardinate node and an instance of this class share all the node interfaces
+   * such as `rclcpp::node_interfaces::NodeParameterInterface`.
+   * Suboardinate nodes are primarily used to organize namespaces and provide a
+   * hierarchical structure, but they are not meant to be completely independent nodes.
```

Comment:
i think that we explicitly add this description to avoid confusion about sub-node behavior.